scripts/triggers-end-to-end-tests/run.spec.js

```diff
@@ -154,6 +160,23 @@ TriggerEndToEndTest.prototype.writeToPubsub = function writeToPubsub(done) {
   return this.invokeHttpFunction("writeToPubsub", done);
 };
 
+TriggerEndToEndTest.prototype.waitForCondition = function(conditionFn, timeout, callback) {
+  let elapsed = 0;
+  let interval = 10;
+  const id = setInterval(() => {
+    if (conditionFn()) {
+      clearInterval(id);
+      callback();
```

Comment:
I don't think it matters, but if `interval` were very large we might evaluate the condition to be true and then throw an error on line 175. It's not wrong or anything I just want to make sure that was your intent.